spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/LazyWeightedServiceInstanceList.java

```diff
@@ -82,77 +83,61 @@ static int greatestCommonDivisor(int a, int b) {
 		return a;
 	}
 
-	static class InterleavedWeightedServiceInstanceSelector {
-
-		static final int MODE_LIST = 0;
-
-		static final int MODE_QUEUE = 1;
-
-		final ServiceInstance[] instances;
-
-		final int[] weights;
-
-		final int greatestCommonDivisor;
-
-		final Queue<Entry> queue;
+	static class O1ServiceInstanceSelector {
```

Comment:
Maybe rename to sth like `WeightedInstanceSelector` as this puts focus on the main feature.